vscode-icon.el

```diff
@@ -176,45 +185,72 @@ Icon Source: https://github.com/vscode-icons/vscode-icons"
                (cdr (assoc file vscode-icon-file-alist))
                (cdr (assoc (file-name-extension file)
                            vscode-icon-file-alist)))))
-        (vscode-icon-if-let* ((filepath (vscode-icon-file-exists-p val)))
+        (vscode-icon-if-let* ((filepath (vscode-icon-file-exists-p
+                                         val light)))
             (vscode-icon-create-image filepath)
           (vscode-icon-default-file))
       (vscode-icon-default-file))))
 
-(defun vscode-icon-file-exists-p (key)
```

Comment:
Could you split the commits up so it's easier to see what's being refactored/re-implemented/added, etc. Hard to tell with lisp-type diffs. :D

Reply:
My bad, the diff is indeed really garbled. I've split the changes to change one function at a time.